laurent22/joplin
# Desktop: Fixes #9597: Fix image rotation not preserved when resizing an attached image

# Summary

- Fixes #9597 by using an [`OffscreenCanvas`](https://developer.mozilla.org/en-US/docs/Web/API/OffscreenCanvas) to resize large images instead of `nativeImage`.
- Adds an automated Playwright test to verify that rotation is preserved.
    - If rotation EXIF metadata is lost, the test image rotates 90 degrees.

# Testing

1. Open the rich text editor and attach `packages/app-desktop/integration-tests/resources/large-jpg-image-with-exif-rotation.jpg`
2. Click "Yes" in the resize dialog
    - Verify that the image isn't rotated
2. Run the automated Playwright tests (verify that they pass)
3. Revert changes to `shim-init-node.ts`
4. Rerun the automated Playwright tests (verify that the rotation-related test fails)

This has been successfully tested on Ubuntu 23.10.
